workos/directory_sync.py

```diff
@@ -52,7 +70,7 @@ def list_users(
         before: Optional[str] = None,
         after: Optional[str] = None,
         order: PaginationOrder = "desc",
-    ) -> SyncOrAsyncListResource: ...
+    ) -> SyncOrAsync[DirectoryUsersListResource]: ...
```

Comment:
Something about this seems odd/duplicative. The list resource is defined as a union:
```
DirectoryUsersListResource = Union[
    AsyncWorkOsListResource[
        DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata
    ],
    WorkOsListResource[DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata],
]
```

`SyncOrAsync` is a convenience to `Union` a type with its `Awaitable` version: `SyncOrAsync = Union[T, Awaitable[T]]`

But both `AsyncWorkOsListResource` and `WorkOsListResource` aren't meant to be `Awaitable`, are they? That's only `AsyncWorkOsListResource`.

In the end this expands to:
```
Union[
  Union[
    AsyncWorkOsListResource[
        DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata
    ],
    WorkOsListResource[DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata],
  ],
  Awaitiable[
    Union[
      AsyncWorkOsListResource[
        DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata
      ],
      WorkOsListResource[DirectoryUserWithGroups, DirectoryUserListFilters, ListMetadata],
    ]
  ]
]
```

I don't think that's what we want.

Reply:
Oh yeah, this is all messed up. Fixed this up a bit with the pagination refactor. I'm bringing those updates in here.